key/key.go

```diff
@@ -0,0 +1,23 @@
+package key
+
+import (
+	logger "github.com/ipfs/go-log"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+var log = logger.Logger("key")
+
+// ID identifies a key, usually it's determined by hash-of-publickey
+type ID = peer.ID
+
+// Public is the public half of a cryptographic key, plus an identifier
+type Public struct {
+	ID  ID
+	Key crypto.PubKey
+}
```

Comment:
While I'm glad to see a distinct `key.ID` type, I don't think we need this struct. Instead I think we should add a `KeyID` field to `profile.Profile` and require users access the keystore to get keys.

_but_, you can get lots of the value of this struct by adding / keeping `PrivKey` and `PubKey` fields on `profile.Profile`, which will save the overhead of repeated lookups. Given that `ProfileStore` is going to wrap `KeyStore`, we can have the profile store populate whatever key into we need.

We should later add a function to the key package that defines the canonical method for getting a n identifier from a given key, which should lower the need for this struct even more

Reply:
perfect